lexos/managers/file_manager.py

```diff
@@ -1359,59 +1359,31 @@ def get_matrix_deprec(self, use_word_tokens: bool, use_tfidf: bool,
         return count_matrix
 
     def get_class_division_map(self):
+        """:return: a panda frame that contains class division map."""
+        # initialize
+        temp_labels = []
+        temp_classes = []
 
-        # TODO: get rid of this horrible function
-        """
-        Args:
-
-        Returns:
-            division_map:
-            name_map:
-            class_label_map:
-        """
-        # create division map
-        division_map = [[0]]  # initialize the division map (at least one file)
+        # active files labels and classes
         files = self.get_active_files()
-        try:
-            # try to get temp label
-            name_map = [[request.form["file_" + str(files[0].id)]]]
-        except KeyError:
-            try:
-                name_map = [[files[0].label]]  # user send a get request.
-            except IndexError:
-                return []  # there is no active file
-        class_label_map = [files[0].class_label]
-
-        # because 0 is defined in the initialize
-        for file_id in range(1, len(files)):
-
-            inside_existing_group = False
-
-            for i in range(len(division_map)):  # for group in division map
-                for existing_id in division_map[i]:
-                    if files[existing_id].class_label == \
-                            files[file_id].class_label:
-                        division_map[i].append(file_id)
-                        try:
-                            # try to get temp label
-                            name_map[i].append(
-                                request.form["file_" + str(files[file_id].id)])
-                        except KeyError:
-                            name_map[i].append(files[file_id].label)
-                        inside_existing_group = True
-                        break
-
-            if not inside_existing_group:
-                division_map.append([file_id])
-                try:
-                    # try to get temp label
-                    name_map.append(
-                        [request.form["file_" + str(files[file_id].id)]])
-                except KeyError:
-                    name_map.append([files[file_id].label])
-                class_label_map.append(files[file_id].class_label)
-
-        return division_map, name_map, class_label_map
+        for file in files:
+            temp_labels.append(file.label)
+            if file.class_label not in temp_classes:
+                temp_classes.append(file.class_label)
+        label_length = len(temp_labels)
+        class_length = len(temp_classes)
+
+        # initialize class division map
+        division_map = pd.DataFrame(
+            data=np.zeros((class_length, label_length), dtype=bool),
+            index=temp_classes,
+            columns=temp_labels)
+
+        # set correct boolean value for each file
+        for file in files:
+            division_map[file.label][file.class_label] = True
```

Comment:
Okay, but a little unintuitive, can you find a way without initializing the data frame? 

Construct it directly